tests/lax_numpy_indexing_test.py

```diff
@@ -982,6 +982,42 @@ def testSegmentSum(self):
     expected = onp.array([13, 2, 7, 4])
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_val={}_idx={}".format(
+          jtu.format_shape_dtype_string(data_shape, val_dtype),
+          jtu.format_shape_dtype_string(data_shape, idx_dtype)),
+       "val_dtype": val_dtype, "idx_dtype": idx_dtype,
+       "rng_idx_factory": rng_idx_factory,
+       "rng_factory": rng_factory,
+       "data_shape": data_shape
+      }
+      for idx_dtype in int_dtypes
+      for val_dtype in default_dtypes
+      for data_shape in [(3,), (3, 4), (3, 4, 4)]
+      for rng_idx_factory in [partial(jtu.rand_int, high=data_shape[0])]
+      for rng_factory in [jtu.rand_default]))
+  def testSegmentMax(self, val_dtype, idx_dtype, rng_idx_factory, rng_factory,
+                     data_shape):
+    def args_maker():
+      rng = rng_factory(onp.random.RandomState(0))
```

Comment:
You should instantiate these outside `args_maker`.

And I think you could instantiate both with `self.rng()`, which is a random state deterministically chose based on the full test case name (including any parameters, etc.)